dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -429,17 +429,85 @@ endif::[]
 [[using-nodeport]]
 == Using a Service NodePort
 
-Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster.
-NodePorts are in the 30-60k range by default, which means a NodePort is
-unlikely to match a service's intended port (for example, 8080 may be exposed
-as 31020). This use of ports is wasteful of scarce port resources.
-However, it is slightly easier to set up. Again, this requires more privileges.
+Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster. You can specify the desired nodePort. When a nodePort is not specified, it is set to a
+port number between `30000` - `60000` by default. Therefore, a NodePort is
+unlikely to match a service's intended port (for example, `8080` may be exposed
+as `31020`). This use of ports is wasteful of scarce port resources. However, it
```

Comment:
I wouldn't call this wasteful.  Also, the default node port range in OpenShift is 30000-32000 (but admins can widen it).

I would say:

> NodePorts are intended to allow load balancing across an entire cluster and are used by the LoadBalancer service type when running on a cloud provider.  You may specify a specific nodePort (if not already in use) or allow the system to assign you one.  When the port is assigned, all nodes in the cluster will allow incoming traffic to that port and then direct the traffic from that node to a local or remote endpoint under the service.  By default, OpenShift allocates ports between 30000 and 32000 as NodePorts.  For very high density environments, you may need to increase the range of ports.  OpenShift automatically opens the node port range at install time for firewalls and cloud security groups for new clusters.

Reply:
@smarterclayton how do you increase the range of ports? Nice description.

Reply:
Thanks @smarterclayton for the updated description. Yes the process for increasing range of ports will make a good addition to the docs.